core/domain/topic_domain.py

```diff
@@ -1969,18 +2091,24 @@ def to_dict(self):
         """
         return {
             'topic_id': self.id,
-            'manager_names': user_services.get_human_readable_user_ids(
+            'manager_names': user_services.get_human_readable_user_ids( # type: ignore[no-untyped-call]
                 self.manager_ids),
             'topic_is_published': self.topic_is_published
         }
 
-    def is_manager(self, user_id):
+    def is_manager(
+        self, user_id: str
+    ) -> bool:
         """Checks whether given user is a manager of the topic.
 
         Args:
-            user_id: str or None. Id of the user.
+            user_id: str. Id of the user.
 
         Returns:
             bool. Whether user is a topic manager of this topic.
         """
-        return bool(user_id in self.manager_ids)
+        if user_id:
+            return bool(user_id in self.manager_ids)
+        raise Exception(
+            'Expected user_id value to be a string, '
+                       'received None')
```

Comment:
From where we call this with possible `None`?

Reply:
[Here](https://github.com/oppia/oppia/blob/2f18e7009b68f6c758c84a25824138eb13bbbdd4/core/domain/topic_domain_test.py#L800), in this method, the argument passed in the `is_manager` method may contain `None` as it is getting initialized [here](https://github.com/oppia/oppia/blob/2f18e7009b68f6c758c84a25824138eb13bbbdd4/core/domain/topic_domain_test.py#L52) and if we see the [method](https://github.com/oppia/oppia/blob/2f18e7009b68f6c758c84a25824138eb13bbbdd4/core/tests/test_utils.py#L2112), it is giving `None` when a user id of a given username doesn't exist. So, should I keep `user_id` as Optional?

Reply:
No, set it as str and in the test just make sure it is not `None` using `assert x is not None`

Reply:
I used `assert x is not None`  in tests and removed the if condition from `is_manager` method. please take a look.